### PR TITLE
Drop >= in suggestions to cabal-version (rebase #7104)

### LINKED ISSUE #7088: "To use ... the package needs to specify at least 'cabal-version: >= 1.18'" ◀ should drop >=
**Describe the bug**
If I have:

```
cabal-version: 1.10
```
`cabal check` complains

```
Warning: To use the 'extra-doc-files' field the package needs to specify at
least 'cabal-version: >= 1.18'.
Warning: Hackage would reject this package.
```

After changing to `cabal-version: >= 1.18` it then complains:

```
Warning: Packages relying on Cabal 1.12 or later should specify a specific
version of the Cabal spec of the form 'cabal-version: x.y'. Use
'cabal-version: 1.18'.
```

Okay, apparently Hackage would not reject `cabal-version: >= 1.18`,
but this is annoying, it should just suggest `cabal-version: 1.18` to the user in the first place.

**System information**
 - cabal-install 3.2 (but this is true for all recent releases in memory: 2.x etc)

**Additional info**
Seems it is determined in Cabal/src/Distribution/PackageDescription/Check.hs

### PR DESCRIPTION
**Pull request by ulysses4ever**
This should fix #7088. All credits go to the author of #7104.
---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
I guess for 1.4 the syntax was still ">= 1.4", but that warning should be removed, anyway, since 1.4 is outside our compatibility window, However, since it's outside our compatibility window, we can just as well offer wrong advice temporarily.

@ffaf1: that's one of the examples of old cruft in `cabal check` that it would be great to get rid of. Compatibility window is defined in CONTRIBUTING.md.

**Comment by ulysses4ever** [comment]
I'll happily remove the mentions of 1.4.

**Comment by Mikolaj** [comment]
> I'll happily remove the mentions of 1.4.

To be clear: I don't remember when the standard changed from `>=` to nothing, so we'd need to figure this out, too.

**Review by fgaz — CHANGES_REQUESTED** [review]
Quoting the cabal check warning:

> With 1.10 or earlier, the 'cabal-version' field must use range syntax rather than a simple version number. Use 'cabal-version: >= 1.10'.

Also the support policy in CONTRIBUTING.md does not apply here, we must support all `cabal-version`s (the file format) **forever**, or at least until a hypothetical Hackage reset

**Comment by Mikolaj** [comment]
> Also the support policy in CONTRIBUTING.md does not apply here. We must support all `cabal-version`s (the file format) **forever**, or at least until a hypothetical Hackage reset

Good point. I stand corrected.

**Comment by ulysses4ever** [comment]
I just pushed a new version that leaves alone all `>=`'s when the warning mentions cabal-version <= 1.10. I hope that's what you, guys, suggested. Let me know if something else is needed.

File: Cabal/src/Distribution/PackageDescription/Check.hs [hunk 1]
```diff
@@ -1292,7 +1292,7 @@ checkCabalVersion pkg =
            && isNothing (setupBuildInfo pkg)
            && buildType pkg == Custom) $
       PackageBuildWarning $
-           "Packages using 'cabal-version: >= 1.24' with 'build-type: Custom' "
+           "Packages using 'cabal-version: 1.24' with 'build-type: Custom' "
```
Comment (fgaz) [inline Cabal/src/Distribution/PackageDescription/Check.hs:1295]: There is still a slight problem with this. It should be 1.24 _or higher_. `>=` conveyed that info, but it was syntactically incorrect.

```suggestion
           "Packages using 'cabal-version: 1.24' or higher with 'build-type: Custom' "
```

Comment (ulysses4ever) [reply Cabal/src/Distribution/PackageDescription/Check.hs]: Thanks, done!

Comment (ulysses4ever) [reply Cabal/src/Distribution/PackageDescription/Check.hs]: Oh, sorry, do you think, I should add "or higher" everywhere?

Comment (fgaz) [reply Cabal/src/Distribution/PackageDescription/Check.hs]: Yes please!

Comment (ulysses4ever) [reply Cabal/src/Distribution/PackageDescription/Check.hs]: Done!

**Review by fgaz — APPROVED** [review]
Thanks!

**Review by Mikolaj — APPROVED** [review]

**Comment by Mikolaj** [comment]
This only changes texts. May we backport this to 3.8?

**Comment by Mikolaj** [comment]
Could you backport?

**Comment by ulysses4ever** [comment]
Let's see... @Mergifyio backport 3.8

**Comment by Mikolaj** [comment]
mergify is very fussy, probably permits only the command in the comment and nothing else

**Comment by ulysses4ever** [comment]
@Mergifyio backport 3.8

**Pull request merged**